mycli/sqlcompleter.py

```diff
@@ -369,48 +369,61 @@ def find_matches(text, collection, start_only=False, fuzzy=True, casing=None):
         yields prompt_toolkit Completion instances for any matches found
         in the collection of available completions.
         """
-        last = last_word(text, include='most_punctuations')
-        text = last.lower()
 
-        completions = []
+        if casing == 'auto':
+            casing = 'lower' if text and text[-1].islower() else 'upper'
+
+        text = text.lower()
+
+        def apply_case(kw):
+            if casing is None:
+                return kw
+            if casing == 'upper':
+                return kw.upper()
+            return kw.lower()
+
+        matches = []
 
         if fuzzy:
             regex = '.*?'.join(map(escape, text))
             pat = compile('(%s)' % regex)
             for item in collection:
                 r = pat.search(item.lower())
                 if r:
-                    completions.append((len(r.group()), r.start(), item))
+                    matches.append(
+                        (len(r.group()), r.start(), apply_case(item)))
         else:
             match_end_limit = len(text) if start_only else None
             for item in collection:
                 match_point = item.lower().find(text, 0, match_end_limit)
                 if match_point >= 0:
-                    completions.append((len(text), match_point, item))
-
-        if casing == 'auto':
-            casing = 'lower' if last and last[-1].islower() else 'upper'
+                    matches.append((len(text), match_point, apply_case(item)))
 
-        def apply_case(kw):
-            if casing == 'upper':
-                return kw.upper()
-            return kw.lower()
-
-        return (Completion(z if casing is None else apply_case(z), -len(text))
-                for x, y, z in completions)
+        return matches
 
     def get_completions(self, document, complete_event, smart_completion=None):
         word_before_cursor = document.get_word_before_cursor(WORD=True)
+        text = last_word(word_before_cursor, include='most_punctuations')
+
+        def sorted_completions(matches):
+            # sort by match point, then match length, then item text
+            matches = sorted(list(matches), key=lambda m:
+                             (m[1], m[0], m[2].lower().strip('`'), m[2].startswith('`')))
```

Comment:
I think this `strip` is causing the tests to fail. The backticks are used to escape the table name if it happens to be a keyword. For instance if someone names the table `table` then it going to conflict with the keyword since `table` is a keyword. This adds backticks around it to escape the string to avoid the conflict.

Reply:
See the comment by @rolandwalker on June 13. The idea is that a table/field surrounded by backticks should match the same one without backticks. I've tried the change the tests to work with this.